### ISSUE TITLE
Failed to install @osjs/pam-auth

### ISSUE THREAD

**Issue by khoaakt** [os-js/osjs-pam-auth#8]
I'm trying to install pam-auth on Node 12.16.1 (Ubuntu Server 18.04.4):
I'm getting the following compilation-time error:

```
khoaakt@servercore:~/OS.js$ npm install --save --production @osjs/pam-auth

> authenticate-pam@1.0.2 install /home/khoaakt/OS.js/node_modules/authenticate-pam
> node-gyp rebuild

make: Entering directory '/home/khoaakt/OS.js/node_modules/authenticate-pam/build'
  CXX(target) Release/obj.target/authenticate_pam/authenticate_pam.o
../authenticate_pam.cc: In function ‘void after_doing_auth(uv_work_t*, int)’:
../authenticate_pam.cc:107:87: warning: ‘v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)’ is deprecated [-Wdeprecated-declarations]
   Nan::MakeCallback(Nan::GetCurrentContext()->Global(), Nan::New(m->callback), 1, args);
                                                                                       ^
In file included from ../authenticate_pam.cc:23:0:
../../nan/nan.h:1024:46: note: declared here
   NAN_DEPRECATED inline v8::Local<v8::Value> MakeCallback(
                                              ^~~~~~~~~~~~
../authenticate_pam.cc: In function ‘Nan::NAN_METHOD_RETURN_TYPE Authenticate(Nan::NAN_METHOD_ARGS_TYPE)’:
../authenticate_pam.cc:147:83: error: no matching function for call to ‘v8::Array::Get(v8::Local<v8::String>)’
   Local<Value> res = options->Get(Nan::New<String>("serviceName").ToLocalChecked());
                                                                                   ^

*too long*

 ^~~~~~~~~~~
authenticate_pam.target.mk:111: recipe for target 'Release/obj.target/authenticate_pam/authenticate_pam.o' failed
make: *** [Release/obj.target/authenticate_pam/authenticate_pam.o] Error 1
make: Leaving directory '/home/khoaakt/OS.js/node_modules/authenticate-pam/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/khoaakt/.nvm/versions/node/v12.16.1/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:311:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! System Linux 4.15.0-88-generic
gyp ERR! command "/home/khoaakt/.nvm/versions/node/v12.16.1/bin/node" "/home/khoaakt/.nvm/versions/node/v12.16.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/khoaakt/OS.js/node_modules/authenticate-pam
gyp ERR! node -v v12.16.1
gyp ERR! node-gyp -v v5.0.5
gyp ERR! not ok
npm WARN hyperapp-nestable@1.1.0 requires a peer of hyperapp@1.1.x but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.2 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! authenticate-pam@1.0.2 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the authenticate-pam@1.0.2 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/khoaakt/.npm/_logs/2020-02-22T19_01_09_346Z-debug.log
khoaakt@servercore:~/OS.js$
```

Looks like we've to use Node 10 or older to get rid of compilation error (caused by authenticate-pam module).

**Comment by andersevenrud** [comment]
> Looks like we've to use Node 10 or older to get rid of compilation error (caused by authenticate-pam module).

Indeed. And there's no fix for it atm, so I might have a stab at it myself.

**Comment by andersevenrud** [comment]
Seems like someone has already looked at it, but not finished (or at least not merged)

https://github.com/Rush/node-authenticate-pam/pull/31

**Comment by RossComputerGuy** [comment]
While looking at this issue, I found this comment https://github.com/Ride-The-Lightning/RTL/issues/115#issuecomment-489464434 and so we may need to replace `authenticate-pam` since its not actively developed and has issues. It will be better to replace it than to wait for it to be fixed.

**Comment by andersevenrud** [comment]
This should have been fixed in #9
